### PR TITLE
test(recurring-job/fstrim): test RWO and RWX volume

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue:
- longhorn/longhorn#7768
- longhorn/longhorn#7771

#### What this PR does / why we need it:

Extend `test_recurring_job_filesystem_trim` with pytest.mark.parametrize fixture to test RWO and RWX volumes.

#### Special notes for your reviewer:

None

#### Additional documentation or context

None

### CODE REVIEW

**Comment by c3y1huang** [comment]
Test result: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/6077/

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -490,7 +493,7 @@ def delete_backup_volume(client, volume_name):
     wait_for_backup_volume_delete(client, volume_name)
 
 
-def create_and_check_volume(client, volume_name,
+def create_and_check_volume(client, volume_name, access_mode=ACCESS_MODE_RWO,
                             num_of_replicas=3, size=SIZE, backing_image="",
```
Comment (james-munson) [inline manager/integration/tests/common.py:497]: I'm confused.  Wouldn't this require an added parameter in all those places that call with positional arguments (or maybe a change to use named arguments)?  Such as this one:  https://github.com/longhorn/longhorn-tests/blob/078c17d8ce66d649ce0d5ed68307cfa28549ef17/manager/integration/tests/test_basic.py#L1000

Comment (c3y1huang) [reply manager/integration/tests/common.py:497]: Good catch, you are correct, this indeed messes up the arguments when optional arguments are invoked as positional arguments.

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by yangchiu** [comment]
There are many test cases modified, but only test case `test_recurring_job_filesystem_trim` is run. Maybe another test is required to make sure affected test cases still work.

**Review by james-munson — APPROVED** [review]
Looks good.

**Comment by c3y1huang** [comment]
> There are many test cases modified, but only test case `test_recurring_job_filesystem_trim` is run. Maybe another test is required to make sure affected test cases still work.

The refactor involves assigning names to the optional argument when they are called. There isn't any behavioral change.

**Comment by yangchiu** [comment]
> 
> Could you share the reason you think this change would make the related cases not work as expected?

Like this one https://github.com/longhorn/longhorn-tests/pull/1689#discussion_r1465338260?

**Comment by c3y1huang** [comment]
> Like this one [#1689 (comment)](https://github.com/longhorn/longhorn-tests/pull/1689#discussion_r1465338260)?

It has been addressed. Any other concerns?

**Review by yangchiu — APPROVED** [review]
LGTM

Test result: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/6157/

**Comment by roger-ryao** [comment]
@mergify backport v1.6.x v1.5.x

**Pull request merged**